egs/swbd/s5c/local/rnnlm/tuning/run_lstm_tdnn_b.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# Copyright 2012  Johns Hopkins University (author: Daniel Povey)  Tony Robinson
+#           2017  Hainan Xu
+#           2017  Ke Li
+
+# This script is similar to rnnlm_lstm_tdnn_a except for adding L2 regularization.
+
+# rnnlm/train_rnnlm.sh: best iteration (out of 50) was 46, linking it to final iteration.
+# rnnlm/train_rnnlm.sh: train/dev perplexity was 38.6 / 45.8. 
+# Train objf: -472.10 -4.26 -4.13 -4.06 -4.01 -3.98 -3.94 -3.92 -3.90 -3.88 -3.87 -3.85 -3.83 -3.82 -3.81 -3.81 -3.79 -3.78 -3.78 -3.77 -3.77 -3.75 -3.75 -3.75 -3.74 -3.74 -3.73 -3.72 -3.72 -3.71 -3.71 -3.70 -3.70 -3.70 -3.69 -3.69 -3.69 -3.68 -3.68 -3.68 -3.68 -3.67 -3.67 -3.66 -3.66 -3.66 -3.65 -3.65 -3.65 -3.65 
+# Dev objf:   -10.65 -4.63 -4.38 -4.26 -4.20 -4.16 -4.12 -4.09 -4.07 -4.04 -4.03 -3.99 -3.98 -3.99 -3.96 -3.94 -3.94 -3.94 -3.92 -3.91 -3.91 -3.90 -3.89 -3.89 -3.88 -3.88 -3.87 -3.87 -3.86 -3.86 -3.85 -3.85 -3.85 -3.85 -3.85 -3.84 -3.84 -3.85 -3.84 -3.84 -3.83 -3.83 -3.83 -3.83 -3.83 -3.83 -3.82 -3.83 -3.83 -3.83 
+
+# Begin configuration section.
+cmd=run.pl
+dir=exp/rnnlm_lstm_tdnn_b
+embedding_dim=1024
+lstm_rpd=256
+lstm_nrpd=256
+embedding_l2=0.001 # embedding layer l2 regularize
+comp_l2=0.001 # component-level l2 regularize
+output_l2=0.001 # output-layer l2 regularize
```

Comment:
I know that in ASR, Dan used smaller l2_reg for output layers to allow larger weights for output layer, but in all of these scripts, output_l2 and comp_l2 are the same values.
Is there any specific reason to have two variables?
Did you get degradation by reducing output_l2?

Reply:
Oh because at the beginning I have thought they may be different so I kept using two variables. I tried smaller output_l2 (than comp_l2) on AMI and the performance degrades. For other datasets, I just tried to match the norms of parameters with those of the best results that I used proportional shrink. So I didn't separately tune the output_l2 much.  I think I can have a try on one or two datasets.